exporters/prometheus/exporter.go

```diff
@@ -344,3 +376,27 @@ func sanitizeName(n string) string {
 
 	return b.String()
 }
+
+func validateMetrics(name, description string, metricType *dto.MetricType, mfs map[string]*dto.MetricFamily) (drop bool, help string) {
+	emf, exist := mfs[name]
+	if !exist {
+		mfs[name] = &dto.MetricFamily{
+			Name: proto.String(name),
+			Help: proto.String(description),
+			Type: metricType,
+		}
+		return false, ""
+	}
+	if emf.GetHelp() != description {
+		if emf.GetType() != *metricType && (emf.GetType() == dto.MetricType_HISTOGRAM || *metricType == dto.MetricType_HISTOGRAM) {
```

Comment:
Can you add some commentary on what this is doing and why?  It's a bit hard to follow the rationale.  Also, it looks like there's no test case that hits this conditional.

Reply:
My bad, I will supply with comments to explain this logic.  There exists one test case for this condition, https://github.com/open-telemetry/opentelemetry-go/pull/3469/files#diff-cca1752451cc515360f89a8e36c9c59ae41e131914eb5171a28ee42a2b3d9168R407-R417
but this case is also hard to understand, I will also enhance this. @Aneurysm9

Reply:
My bad, after checking the logic again, I think this condition could be removed. 
The reason of this logic is, when I first commit the code, I use instrument name as the key of cache `metricFamilies`. After the suggestion of @dashpole 
https://github.com/open-telemetry/opentelemetry-go/pull/3469#discussion_r1029431899
I change the key to `c.getName(m)`, this condition is not needed anymore.